pkg/config/config.go

```diff
@@ -339,10 +369,54 @@ func validate(config *Config) error {
 	if config.MinimumMinReplicas >= int(config.MaximumMinReplicas) {
 		return fmt.Errorf("MinimumMinReplicas should be less than MaximumMinReplicas")
 	}
-	if config.MaximumMinReplicas > config.MaximumMaxReplicas {
+
+	// Find the minimum value of MaximumMaxReplicas across all service groups
+	minOfMaximumMaxReplicas := config.MaximumMaxReplicas // Start with the default value of MaximumMaxReplicas
+	for _, group := range config.MaximumMaxReplicasPerService {
+		if group.MaximumMaxReplica < minOfMaximumMaxReplicas {
+			minOfMaximumMaxReplicas = group.MaximumMaxReplica
+		}
+	}
+
+	// Check for non-negative values
+	if minOfMaximumMaxReplicas < 0 {
+		return fmt.Errorf("MaximumMaxReplicas should contain non-negative values")
+	}
+
+	// Ensure ServiceGroupNames in MaximumMaxReplicas match defined ServiceGroups
+	serviceGroupMap := make(map[string]bool)
+	for _, sg := range config.ServiceGroups {
+		serviceGroupMap[sg.Name] = true
+	}
+
+	for _, maxReplicas := range config.MaximumMaxReplicasPerService {
+		if maxReplicas.ServiceGroupName != "" {
+			if _, exists := serviceGroupMap[maxReplicas.ServiceGroupName]; !exists {
+				return fmt.Errorf("ServiceGroupName %s in MaximumMaxReplicas is not defined in ServiceGroups", maxReplicas.ServiceGroupName)
+			}
+		}
+	}
+
+	// Ensure no duplicates in ServiceGroups
+	seenServiceGroups := make(map[string]bool)
+	for _, sg := range config.ServiceGroups {
+		if seenServiceGroups[sg.Name] {
+			return fmt.Errorf("duplicate ServiceGroupName found: %s", sg.Name)
+		}
+		seenServiceGroups[sg.Name] = true
+	}
+
+	// Check all entries in MaximumMaxReplicasPerService have non-nil ServiceGroupName
+	for _, maxReplicas := range config.MaximumMaxReplicasPerService {
+		if maxReplicas.ServiceGroupName == "" {
+			return fmt.Errorf("ServiceGroupName should not be nil in MaximumMaxReplicasPerService entries")
+		}
+	}
+
+	if config.MaximumMinReplicas > minOfMaximumMaxReplicas {
 		return fmt.Errorf("MaximumMinReplicas should be less than or equal to MaximumMaxReplicas")
 	}
-	if config.PreferredMaxReplicas >= int(config.MaximumMaxReplicas) {
+	if config.PreferredMaxReplicas >= int(minOfMaximumMaxReplicas) {
```

Comment:
I think `config.MaximumMaxReplicas` and `minOfMaximumMaxReplicas` should be validated with `PreferredMaxReplicas` and `MaximumMinReplicas` separately for better error messages.